runtime/queries/metricsview_rows.go

```diff
@@ -235,7 +236,11 @@ func (q *MetricsViewRows) buildMetricsRowsSQL(mv *runtimev1.MetricsView, dialect
 			panic("timeRollupColumnName is set, but time dimension info is missing")
 		}
 
-		rollup := fmt.Sprintf("date_trunc('%s', %s) AS %s", convertToDateTruncSpecifier(q.TimeGranularity), safeName(mv.TimeDimension), safeName(timeRollupColumnName))
+		timezone := "UTC"
+		if q.TimeZone != "" {
+			timezone = q.TimeZone
+		}
+		rollup := fmt.Sprintf("time_bucket(INTERVAL '%s', %s::TIMESTAMPTZ, '%s') AS %s", convertToTimeBucketSpecifier(q.TimeGranularity), safeName(mv.TimeDimension), timezone, safeName(timeRollupColumnName))
```

Comment:
Also need to guard against SQL injection – in this case, doing `args = append(arsg, timezone)` should be straightforward